src/celeritas/global/detail/ActionSequence.cc

```diff
@@ -89,11 +90,13 @@ void ActionSequence::begin_run(CoreParams const& params, CoreState<M>& state)
 template<MemSpace M>
 void ActionSequence::execute(CoreParams const& params, CoreState<M>& state)
 {
+    ScopedProfiling profile_this{"step"};
     if (M == MemSpace::host || options_.sync)
     {
         // Execute all actions and record the time elapsed
         for (auto i : range(actions_.size()))
         {
+            ScopedProfiling profile_this{actions_[i]->label()};
```

Comment:
Should we possibly enable profiling only for one or the other of these? Is any of the profiling affected by the synchronize call?

Reply:
synchronize shouldn't change anything for Nsight compute since we're only looking at the kernel execution on the device, for Nisght system we'd definitely see each iteration take longer